.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1370,7 +1368,7 @@ private DateTimeResolutionResult ParseWeekOfYear(string text, DateObject referen
                 targetWeekMonday = lastDayWeekMonday;
                 weekNum = Cal.GetWeekOfYear(targetWeekMonday, CalendarWeekRule.FirstFourDayWeek, DayOfWeek.Monday);
 
-                ret.Timex = $"{year.ToString("D4")}-{targetWeekMonday.Month.ToString("D2")}-W{weekNum}";
+                ret.Timex = $"{year:D4}-{targetWeekMonday.Month:D2}-W{weekNum}";
```

Comment:
I'd prefer if the generation of timex expressions goes to it's own method.

Reply:
I agree, but there is a lot of code of this kind need to be refactored, we can arrange a sperate PR to fix it.